src/multibody/model.hpp

```diff
@@ -158,20 +160,46 @@ namespace se3
                         );
 
     ///
+    /// \brief Add a joint to the frame tree.
+    ///
+    /// \param[in] jointIndex Index of the joint.
+    /// \param[in] frameIndex Index of the parent frame. If negative,
+    ///            the parent frame is the frame of the parent joint.
+    ///
+    /// \return The index of the new frame.
+    ///
+    FrameIndex addJointFrame (const JointIndex& jointIndex,
+                                    int         frameIndex = -1);
+
+    ///
     /// \brief Append a body to a given joint of the kinematic tree.
     ///
     /// \remark This method also adds a Frame of same name to the vector of frames.
     ///
     /// \param[in] joint_index Index of the supporting joint.
     /// \param[in] Y Spatial inertia of the body.
     /// \param[in] body_placement The relative placement of the body regarding to the parent joint. Set default to the Identity placement.
-    /// \param[in] body_name Name of the body. If empty, the name is random.
     ///
     /// \sa Model::addJoint
     ///
     void appendBodyToJoint(const JointIndex joint_index, const Inertia & Y,
-                           const SE3 & body_placement = SE3::Identity(),
-                           const std::string & body_name = "");
+                           const SE3 & body_placement = SE3::Identity());
+
+    ///
+    /// \brief Add a body to the frame tree.
+    ///
+    /// \param[in] body_name Name of the body.
+    /// \param[in] parentJoint Index of the parent joint.
+    /// \param[in] body_placement The relative placement of the body regarding to the parent joint. Set default to the Identity placement.
+    /// \param[in] previousFrame Index of the parent frame. If negative,
+    ///            the parent frame is the frame of the parent joint.
+    ///
+    /// \return The index of the new frame.
+    ///
+    FrameIndex addBodyFrame (const std::string& body_name,
```

Comment:
It is a helper. It keeps the API more consistent (`addJoint` / `addJointFrame`, `appendBodyToJoint` / `addBodyFrame`).

I find it more convenient to do:
```cpp
idx = model.addJoint(...);
model.addJointFrame(idx);
model.appendBodyToJoint(...);
model.addBodyFrame("body_name", idx);
```
instead of:
```cpp
idx = model.addJoint(...);
/* parentFrameId = */ model.addJointFrame(idx);
model.appendBodyToJoint(...);
// compute parentFrameId, something like
// parentFrameId = model.getFrameId("joint_name")
model.addFrame(Frame("body_name", idx, parentFrameId, placement, BODY));
```